test/12_missingarray.jl

```diff
@@ -216,13 +216,16 @@ const ≅ = isequal
                 @test x[3] === CategoricalValue(x.pool, 3)
                 @test levels(x) == ["e", "a", "b", "c"]
 
+                # check that x is restored correctly when dropping levels is not allowed
                 @test_throws ArgumentError levels!(x, ["e", "c"])
+                @test x == ["c", "b", "b"]
+                @test levels(x) == ["e", "a", "b", "c"]
+
                 @test levels!(x, ["e", "c"], allowmissing=true) === x
                 @test levels(x) == ["e", "c"]
                 @test x[1] === CategoricalValue(x.pool, 2)
                 @test x[2] === missing
                 @test x[3] === missing
-                @test levels(x) == ["e", "c"]
```

Comment:
Also apply these changes to 11_array.jl.

Reply:
done. I've addressed your latest review (thanks!) and squashed the commits.